File: 11/Zdrojove kody/Hackathon2023/Server/DBModels/Place.cs
namespace Server.DBModels
{
    public class Place
    {
        public Place(int placeID, string name, string web, int addressID, string description, string imageUrl)
        {
            PlaceID = placeID;
            Name = name;
            Web = web;
            AddressID = addressID;
            Description = description;
            ImageUrl = imageUrl;
        }

        public Place(int placeID, string name, string web, string description)
        {
            PlaceID = placeID;
            Name = name;
            Web = web;
            Description = description;
        }

        public int PlaceID { get; private set; }
        public string Name { get; private set; }
        public string Web { get; private set; }
        public int AddressID { get; set; }
        public string Description { get; private set; }
        public string ImageUrl { get; set; }
    }
}

File: 11/Zdrojove kody/Hackathon2023/frontend/types/Place.ts
export interface Place {
  PlaceID: number;
  Name: string;
  Web: string;
  Address: {
    Street: string;
    City: string;
    Zip: string;
    Longitude: number;
    Latitude: number;
  };
  Description: string;
  ImageUrl: string;
  Likes: number;
  Dislikes: number;
  Comments: {
    Name: string;
    // Avatar?: string;
    Text: string;
  }[];
}

export enum Liked {
  Liked = 1,
  Disliked = -1,
  NoReaction = 0,
}
